storage/aws/issuers.go

```diff
@@ -62,31 +65,66 @@ func (s *IssuersStorage) keyToObjName(key []byte) string {
 }
 
 // AddIssuers stores Issuers values under their Key if there isn't an object under Key already.
+// Wrap with a local cache to avoid unecessary read and write operations as issuers are reused
+// accross chains.
 func (s *IssuersStorage) AddIssuersIfNotExist(ctx context.Context, kv []storage.KV) error {
-	// We first try and see if this issuer cert has already been stored since reads
-	// are cheaper than writes.
+	errG := errgroup.Group{}
 	for _, kv := range kv {
-		objName := s.keyToObjName(kv.K)
-		put := &s3.PutObjectInput{
-			Bucket:      aws.String(s.bucket),
-			Key:         aws.String(objName),
-			Body:        bytes.NewReader(kv.V),
-			ContentType: aws.String(s.contentType),
-			IfNoneMatch: aws.String("*"),
-		}
+		errG.Go(func() error {
+			objName := s.keyToObjName(kv.K)
+			put := &s3.PutObjectInput{
+				Bucket:      aws.String(s.bucket),
+				Key:         aws.String(objName),
+				Body:        bytes.NewReader(kv.V),
+				ContentType: aws.String(s.contentType),
+				IfNoneMatch: aws.String("*"),
+			}
+
+			// If we run into a precondition failure error, check that the object
+			// which exists contains the same content that we want to write.
+			// If so, we can consider this write to be idempotently successful.
+			if _, err := s.s3Client.PutObject(ctx, put); err != nil {
+				var apiErr smithy.APIError
+				if errors.As(err, &apiErr); apiErr.ErrorCode() == "PreconditionFailed" {
```

Comment:
I think this is probably fine, but _technically_ should it be `if errors.As(err, &apiErr) && apiErr.ErrorCode() == ...` in order to not use uninitialised contents of `apiErr` if it's not the right sort of error?

In practice, unless "PreconditionFailed" is the default value for `ErrorCode()` on an uninitialised `APIError` it probably boils down to the same result - hitting the return on L101, but it triggered me nonetheless :)